### PR TITLE
Add JDK17 support

### LINKED ISSUE #14: Cannot set Java version to 17
### Troubleshooting

- [x] I've searched [discuss.bitrise.io](https://discuss.bitrise.io/) for possible solutions.
- Which version of the step is effected? 1.0.3
- Is the issue reproducible with the latest version? YES
- Does the issue happen sporadically, or every time? EVERY TIME
- Is the issue reproducible locally by following our [local debug guide](https://devcenter.bitrise.io/troubleshooting/debugging-your-build-on-your-own-machine/)? NOT APPLICABLE

### Useful information

- [FAQ](https://devcenter.bitrise.io/faq/faq-index/)
- [devcenter.bitrise.io](https://devcenter.bitrise.io/)

### Issue description

The Ubuntu image, on Bitrise, has JDK 17 pre-installed.

```
root@prdfree-20-04-use1c-220309085745-grybk9t8v6i8vbepvof986:/bitrise/src# ./gradlew -q javaToolchains

 + Options
     | Auto-detection:     Enabled
     | Auto-download:      Enabled

 + Private Build JDK 1.8.0_312-8u312-b07-0ubuntu1~20.04-b07
     | Location:           /usr/lib/jvm/java-8-openjdk-amd64
     | Language Version:   8
     | Vendor:             Private Build
     | Is JDK:             true
     | Detected by:        Common Linux Locations

 + Private Build JDK 17.0.1+12-Ubuntu-120.04
     | Location:           /usr/lib/jvm/java-17-openjdk-amd64
     | Language Version:   17
     | Vendor:             Private Build
     | Is JDK:             true
     | Detected by:        Common Linux Locations

 + Ubuntu JDK 11.0.13+8-Ubuntu-0ubuntu1.20.04
     | Location:           /usr/lib/jvm/java-11-openjdk-amd64
     | Language Version:   11
     | Vendor:             Ubuntu
     | Is JDK:             true
     | Detected by:        Common Linux Locations

 + Invalid toolchains
     + /usr/lib/jvm/openjdk-17
       | Error:              A problem occurred starting process 'command '/usr/lib/jvm/openjdk-17/bin/java''
     + /usr/lib/jvm/openjdk-11
       | Error:              A problem occurred starting process 'command '/usr/lib/jvm/openjdk-11/bin/java''
```

Therefore, I expect the `set-java-version` step to allow me to select JDK 17.

But the step currently only supports JDK 8 and 11:

### Bitrise info

- Build URL: _Please share the build URL with us._
- [Bitrise Support](https://devcenter.bitrise.io/troubleshooting/enabling-bitrise-support-user/) enabled: _YES / NO_
- Log: 

<details>

```
+------------------------------------------------------------------------------+
| (1) Select JDK version                                                       |
+------------------------------------------------------------------------------+
| id: set-java-version                                                         |
| version: 1.0.3                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2022-03-09T09:15:19Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
failed to parse config:
- JavaVersion: 17: value is not in value options (opt[11,8])
Input:
- JavaVersion: 
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | Select JDK version (exit code: 1)                             | 1.16 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...b.com/bitrise-steplib/bitrise-step-set-java-version/issues |
| Source: https://github.com/bitrise-steplib/bitrise-step-set-java-version     |
+---+---------------------------------------------------------------+----------+
```

</details>

### Steps to reproduce

1. Use this step:

```
  - set-java-version@1:
      inputs:
        - set_java_version: 17
```
2. Run a build

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

Adding the option to activate JDK17, as it is now preinstalled to the stacks.

Also: closes #14 

### Changes

- Update `go-utils` and `go-steputils` dependencies to v2
- Clean up some stuff here and there
- Improve step description and inputs/outputs
- Add new input value for JDK17

### CODE REVIEW

File: javasetter/javasetter.go [hunk 1]
```diff
@@ -69,7 +67,7 @@ func (j JavaSetter) SetJava(version JavaVersion) (Result, error) {
 
 func (j JavaSetter) setJavaMac(version JavaVersion) (Result, error) {
 	if version == JavaVersion8 {
-		version = JavaVersion("1.8")
```
Comment (ofalvai) [inline javasetter/javasetter.go:72]: redundant string conversion

File: javasetter/javasetter.go [hunk 2]
```diff
@@ -104,18 +102,19 @@ func (j JavaSetter) setJavaMac(version JavaVersion) (Result, error) {
...
-			mainDir := "/usr/lib/jvm/java-8-openjdk-amd64"
-			return mainDir+"/jre/bin/java", mainDir+"/bin/javac", mainDir+"/bin/javadoc", mainDir
-		case JavaVersion11:
-			mainDir := "/usr/lib/jvm/java-11-openjdk-amd64"
-			return mainDir+"/bin/java", mainDir+"/bin/javac", mainDir+"/bin/javadoc", mainDir
-		default:
-			return "", "", "", ""
-		}
-	}()
+	var javaHome string
```
Comment (ofalvai) [inline javasetter/javasetter.go:105]: This whole code block was a lot more complex than necessary. The key here is that JDK8 had a unique `java` path, but I double-checked it, and `$JAVA_HOME/bin/java`is also present in the JDK8 folder (it's a symlink to `$JAVA_HOME/jre/bin/java`). This means that the relative paths can be shared between all 3 cases, the only unique variable is the JAVA_HOME.

File: step.go [hunk 3]
```diff
@@ -109,7 +101,8 @@ func (j JavaSelector) Export(result javasetter.Result) error {
 	j.logger.Println()
 	j.logger.Infof("Export step outputs")
 	j.logger.Printf("- Exporting JAVA_HOME=%s", result.JavaHome)
-	if err := tools.ExportEnvironmentWithEnvman("JAVA_HOME", result.JavaHome); err != nil {
```
Comment (ofalvai) [inline step.go:112]: Accidental v1 `go-steputils` usage

File: step.go [hunk 4]
```diff
@@ -82,11 +76,13 @@ func (j JavaSelector) printJavaCVersion() error {
 	return err
 }
 
-// Run ...
 func (j JavaSelector) Run(cfg Config) (javasetter.Result, error) {
 	versionToSet := javasetter.JavaVersion(cfg.javaVersion)
 	setter := javasetter.New(j.logger, j.cmdFactory)
 	result, err := setter.SetJava(versionToSet)
+	if err != nil {
```
Comment (ofalvai) [inline step.go:83]: Missing error handling 👀

**Review by godrei — APPROVED** [review]

**Pull request merged**